src/sqlfluff/core/templaters/slicers/tracer.py

```diff
@@ -197,7 +206,8 @@ def __init__(self, raw_str: str, env: Environment):
 
         # Internal bookkeeping
         self.slice_id: int = 0
-        self.inside_set_or_macro: bool = False
+        self.inside_set_or_macro: bool = False  # {% set %} or {% macro %}
+        self.inside_block = False  # {% block %}
```

Comment:
Do these need to be treated differently? Or could it be `self.inside_set_or_macro_or_block`? I'm guessing so from above code but thought I'd ask the question, since it's similar elsewhere.

Reply:
Yes, it has to be different. `{% block %}` is rendered into the output at both the point of _definition_ and the point of _use_, while `{% set %}` and `{% macro %}` are only rendered at the point of _use_.